src/client/Microsoft.Identity.Client/Platforms/Android/AndroidBrokerHelper.cs

```diff
@@ -216,6 +207,58 @@ public void CheckForBrokerAccountInfoInAccountManager(IDictionary<string, string
             throw new MsalUiRequiredException(MsalError.NoAndroidBrokerAccountFound, MsalErrorMessage.NoAndroidBrokerAccountFound);
         }
 
+        private string GetBrokerAccounts(IDictionary<string, string> brokerPayload, Activity callerActivity)
+        {
+            Bundle getAccountsBundle = GetBrokerAccountBundle(brokerPayload);
+            getAccountsBundle.PutString(BrokerConstants.BrokerAccountManagerOperationKey, BrokerConstants.GetAccounts);
+
+            //This operation will acquire all of the accounts in the account manager for the given client ID
+            var result = _androidAccountManager.AddAccount(BrokerConstants.BrokerAccountType,
+                BrokerConstants.AuthtokenType,
+                null,
+                getAccountsBundle,
+                null,
+                null,
+                GetPreferredLooper(callerActivity));
+
+            Bundle bundleResult = (Bundle)result?.Result;
+            return bundleResult?.GetString(BrokerConstants.BrokerAccounts);
+        }
+
+        /// <summary>
+        /// This method will acquire all of the accounts in the account manager that have an access token for the given client ID.
+        /// </summary>
+        public IEnumerable<IAccount> GetBrokerAccountsInAccountManager(IDictionary<string, string> brokerPayload)
+        {
+            var accounts = GetBrokerAccounts(brokerPayload, null);
+
+            if (string.IsNullOrEmpty(accounts))
+            {
+                _logger.Info("Android account manager didn't return any accounts.");
+            }
+
+            List<IAccount> brokerAccounts = new List<IAccount>();
+
+            if (!string.IsNullOrEmpty(accounts))
+            {
+                dynamic authResult = JArray.Parse(accounts);
```

Comment:
Why do you need this `dynamic` ? We have better ways of performing JSON deserialization. 

When you do `Json.Deserialize<T>()`, reflection is only used once. Subsequent calls to Deserialize<T> will no longer reflect on the type T. 

I believe this optimization is bypassed if you use `dynamic`

Reply:
The JSON deserialization methods we have now do not work on Android. Whenever i tried to use them, they resulted in a null value. dynamic is the only thing that I could get to work

Reply:
Interesting behavior. I am curious to find out the reason for that.

Reply:
There are many other places where we deserialize using Deserialize<T>() in Android, for example when communicating with EVO. Newtonsoft.Json definitely works on Android....

Reply:
I would create an issue to fix this and at least document.

Reply:
Well, this piece of code is not perf sensitive (public client flows generally are not), so not sure we can log a perf bug. Otherwise, this has the cool benefit that it works :) and [refactoring, not on the backlog](https://ronjeffries.com/xprog/articles/refactoring-not-on-the-backlog/).